run-dev: Suppress the notices by third-party tools.

ganpa3 opened the pull request:
Suppress notices made by third-party tools such as
webpack to limit the `run-dev.py` startup output.
And made the terminal message a bit more clear about
accessing the server.

Fixes #16846

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->
Ran ```./tools/test-all```

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![webpack](https://user-images.githubusercontent.com/58626718/105514920-0fb48400-5cfa-11eb-82dc-ed5e6b37259d.png)



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
I'd prefer to configure `webpack` to not provide the notices we don't want, rather than sending it's stdout to /dev/null, which would suppress any notice that `webpack` tries to provide.

ganpa3 commented:
@timabbott Sorry for the delay. I have updated the PR. Added the [noinfo](https://webpack.js.org/configuration/dev-server/#devservernoinfo-) property. Can you please review the PR.

ganpa3 commented:
@timabbott would you please review this PR?

timabbott commented:
This is great!  Can we add back a hook to print a message once webpack has finished compiling?  I think that'd be helpful since the server isn't fully available until webpack has finished compiling.

https://webpack.js.org/api/compiler-hooks/#done might be the right API, but I'm not sure.

ganpa3 commented:
@timabbott Done. I kept the message similar to webpack so that it's easier to recognize. Also, updated the screenshot.

timabbott commented:
Merged after a small change to the string, thanks @ganpa3!
